From 73e9e3f008bb0be0ec5b9429f5208b94fac25d93 Mon Sep 17 00:00:00 2001
From: Scott K Logan <logans@cottsay.net>
Date: Thu, 11 Feb 2021 18:39:10 -0800
Subject: [PATCH] Drop membership in rmw_implementation_packages group

Since this package compiles as an empty package due to the absence of
GurumDDS, it shouldn't be considered a valid member of the
rmw_implementation_packages group.

Signed-off-by: Scott K Logan <logans@cottsay.net>
---
 package.xml          | 2 --
 rpm/template.spec.em | 5 +++++
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/package.xml b/package.xml
index 75142a0..c63a13e 100644
--- a/package.xml
+++ b/package.xml
@@ -35,8 +35,6 @@
   <test_depend>ament_lint_auto</test_depend>
   <test_depend>ament_lint_common</test_depend>
 
-  <member_of_group>rmw_implementation_packages</member_of_group>
-
   <export>
     <build_type>ament_cmake</build_type>
   </export>
diff --git a/rpm/template.spec.em b/rpm/template.spec.em
index bebb96b..549f602 100644
--- a/rpm/template.spec.em
+++ b/rpm/template.spec.em
@@ -1,3 +1,8 @@
+@{
+# Drop membership in rmw_implementation_packages group
+Provides.remove('ros-' + Rosdistro + '-rmw-implementation-packages(member)')
+Supplements.remove('ros-' + Rosdistro + '-rmw-implementation-packages(all)')
+}@
 %bcond_without tests
 %bcond_without weak_deps
 
-- 
2.25.1

